src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs

```diff
@@ -97,7 +97,12 @@ public void StartForwarding(ScriptInvocationContext context, Uri httpUri)
             httpContext.Items[ScriptConstants.HttpProxyingEnabled] = bool.TrueString;
 
             // add invocation id as correlation id
-            httpRequest.Headers.TryAdd(ScriptConstants.HttpProxyCorrelationHeader, context.ExecutionContext.InvocationId.ToString());
+            if (!httpRequest.Headers.TryAdd(ScriptConstants.HttpProxyCorrelationHeader, context.ExecutionContext.InvocationId.ToString()))
+            {
+                // try to remove the header if it already exists
+                httpRequest.Headers.Remove(ScriptConstants.HttpProxyCorrelationHeader);
+                httpRequest.Headers.TryAdd(ScriptConstants.HttpProxyCorrelationHeader, context.ExecutionContext.InvocationId.ToString());
+            }
 
             var forwardingTask = _httpForwarder.SendAsync(httpContext, httpUri.ToString(), _messageInvoker, _forwarderRequestConfig).AsTask();
```

Comment:
The only other thing that I was wondering is if it'd make sense to use an `HttpTransformer` instead here. But I think that since this is so straightforward it doesn't buy us much.